### PR TITLE
Numpy add numpy linalg norm and numpy min

### PR DESCRIPTION
**Pull request by gyshi**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here
@haojin2

### CODE REVIEW

**Comment by haojin2** [comment]
@gyshi Please separate this into 2 PRs: one for linalg.norm, the other for np.min.

**Comment by gyshi** [comment]
> @gyshi Please separate this into 2 PRs: one for linalg.norm, the other for np.min.

OK， because i use min in my op linalg.norm, so should I write np.min and test it again in a new PR?

**Comment by gyshi** [comment]
done

File: python/mxnet/ndarray/numpy/linalg.py [hunk 1]
```diff
@@ -19,50 +19,184 @@
...
+    on the value of the ``ord`` parameter.
 
     Parameters
     ----------
     x : ndarray
-        Input array.
-    ord : {'fro'}, optional
-        Order of the norm.
+        Input array.  If `axis` is None, `x` must be 1-D or 2-D.
+    ord : {non-zero int, inf, -inf, 'fro', 'nuc'}, optional
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/linalg.py]: if something ('nuc') is not supported, firstly do not list it as an option.

File: python/mxnet/ndarray/numpy/linalg.py [hunk 2]
```diff
@@ -19,50 +19,184 @@
...
     """
-    if ord is not None and ord != 'fro':
-        raise ValueError('only support Frobenius norm for now, received ord={}'.format(str(ord)))
-    if isinstance(axis, tuple) and len(axis) > 2:
-        raise ValueError('Improper number of dimensions to norm')
-    if ord == 'fro' and x.ndim > 2 and axis is None:
-        raise ValueError('Improper number of dimensions to norm')
-    return _mx_nd_np.sqrt(_mx_nd_np.sum(x * x, axis=axis, keepdims=keepdims))
+    if ord == 'nuc':
+        raise NotImplementedError("norm is not support svd for this operations.")
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/linalg.py]: Please give an easy-to-understand error message

File: python/mxnet/symbol/numpy/_symbol.py [hunk 3]
```diff
@@ -1653,7 +1653,7 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None, axis
       GPU.
     """
     if isinstance(start, (list, _np.ndarray)) or \
-        isinstance(stop, (list, _np.ndarray)):
+            isinstance(stop, (list, _np.ndarray)):
```
Comment (haojin2) [inline python/mxnet/symbol/numpy/_symbol.py]: No need for extra indentation here.

File: src/operator/numpy/np_broadcast_reduce_op_value.cc [hunk 4]
```diff
@@ -77,9 +77,9 @@ NNVM_REGISTER_OP(_backward_np_sum)
 
 inline bool IsIntType(const int dtype) {
   return (dtype == mshadow::kUint8 ||
-          dtype == mshadow::kInt32 ||
-          dtype == mshadow::kInt8 ||
-          dtype == mshadow::kInt64);
+      dtype == mshadow::kInt32 ||
```
Comment (haojin2) [inline src/operator/numpy/np_broadcast_reduce_op_value.cc]: Why making changes here? Did you apply some kind of linter to this?

File: python/mxnet/ndarray/numpy/linalg.py [hunk 5]
```diff
@@ -19,50 +19,184 @@
 
 from __future__ import absolute_import
 from . import _op as _mx_nd_np
+from . import _internal as _npi
+
 
 __all__ = ['norm']
 
 
+# pylint: disable=line-too-long, too-many-return-statements, too-many-branches, no-member, consider-using-in, no-else-return
```
Comment (marcoabreu) [inline python/mxnet/ndarray/numpy/linalg.py]: Please don't just disable these rules.

File: python/mxnet/symbol/numpy/linalg.py [hunk 6]
```diff
@@ -20,49 +20,184 @@
 from __future__ import absolute_import
 from . import _symbol
 from . import _op as _mx_sym_np
+from . import _internal as _npi
 
 __all__ = ['norm']
 
 
+# pylint: disable=line-too-long, too-many-return-statements, too-many-branches, no-member, consider-using-in, no-else-return
```
Comment (marcoabreu) [inline python/mxnet/symbol/numpy/linalg.py]: Same here

File: python/mxnet/ndarray/numpy/linalg.py (hunk 5 above)
Comment (gyshi) [reply python/mxnet/ndarray/numpy/linalg.py]: ok, i will modify my operator after op svd  and min  are merged

**Pull request closed without merging**